unicode_regex.h

```diff
@@ -0,0 +1,252 @@
+#pragma once
+
+#include "unicode.h"
+#include "unordered_set"
+
+class llm_regex {

```

Comment:
No need for this `class` - use functions with `regex_` prefix

Move everything from `unicode_regex.h` into `unicode.h`